Show Guids as Hex and Display hex opcodes

ogmage78 opened the pull request:
Added the optional ability to see the guids in the tree view as hex and added the Hex Opcode to the main window.

tfarley commented on aclogview/CM_Writing.cs:
```diff
@@ -55,30 +55,30 @@ public class CM_Writing : MessageProcessor {
             TreeNode rootNode = new TreeNode(this.GetType().Name);
             rootNode.Expand();
-            rootNode.Nodes.Add("i_objectID = " + i_objectID);
+            rootNode.Nodes.Add("i_objectid = " + i_objectid.ToString("X"));
```
Any reason that a few of these always use hex and don't check to see if UseHex is enabled?

ogmage78 replied:
LOL that would be because I missed them.   I will find them and fix.

ogmage78 replied:
I have made the requested changes.

tfarley commented on aclogview/CM_Combat.cs:
```diff
@@ -72,7 +73,15 @@ public class TargetedMeleeAttack : Message {
+            else
+            {
+                rootNode.Nodes.Add("i_targetID = " + this.i_targetID);
+            }
```
I'm thinking these blocks should be done via a separate function, since this is turning a single line into 8 lines and causes every treeview contributor to reach into globals. What about something like:

rootNode.Nodes.Add("i_targetID = " + formatIDForDisplay(this.i_targetID));

Where formatIDForDisplay would internally check UseHex to either just return the input as a string or "0x" + input.ToString("X"); if UseHex is enabled.

ogmage78 replied:
Way better idea.   I will do that.

tfarley commented:
Added some comments. I'm wondering if we might want to _always_ display ids as hex though, and just do away with UseHex? Not sure what the typical dev workflow looks like - if keeping decimal representations of IDs is of any use or not.

ogmage78 commented:
Well - everyone I talked to uses hex - but I but it in switchable just in case someone liked decimal.   It would be much easier to just do hex.   I am good either way.   Thanks for doing the viewer for us.   It is really helping a bunch.

tfarley commented:
Sorry, looks like merging another PR created some conflicts. Going to leave the decision of checkbox vs. hex-only up to you - whatever you think is most useful.

ogmage78 commented:
I have made the requested changes and fixed the merge conflicts.   Let me know if you see anything else that needs to be addressed.

tfarley commented on aclogview/Program.cs:
```diff
@@ -5,6 +5,39 @@
+            }
+
+            return theValue.ToString();
+        }
```
This is a dupe of FormatGuid in Utility class. You should only have it in one spot.

tfarley commented on aclogview/CM_Communication.cs:
```diff
@@ -252,8 +253,8 @@ public override void contributeToTreeView(TreeView treeView)
-            rootNode.Nodes.Add("SenderName = " + SenderName.m_buffer);
-            rootNode.Nodes.Add("SenderID = " + SenderID);
+            rootNode.Nodes.Add("SenderName = " + SenderName.m_buffer);
+            rootNode.Nodes.Add("SenderID = " + Utility.FormatGuid(this.SenderID));                        
```
There shouldn't be a need to qualify all of these properties with "this."

tfarley commented on aclogview/CM_Movement.cs:
```diff
@@ -489,8 +492,8 @@ public class MovementEvent : Message {
-            rootNode.Nodes.Add("stickToObject = " + stickToObject);
-            rootNode.Nodes.Add("moveToObject = " + moveToObject);
+            rootNode.Nodes.Add("stickToObject = " + stickToObject.ToString("X"));
+            rootNode.Nodes.Add("moveToObject = " + moveToObject.ToString("X"));
```
A bunch of these (not just here) are still not going through a call to FormatGuid.

ogmage78 commented:
Thanks for catching those errors.   I need to quit coding when I am tired.   :)   I removed the duplicate method - that was left in unintentionally.   I updated the areas were I had missed using the new method.

Forbiddenz commented:
Cant wait for this PR, make it happen guys! 
HEX!!!!

Good job!

tfarley commented on aclogview/Form1.cs:
```diff
@@ -112,9 +113,11 @@ protected override void OnClosing(CancelEventArgs e)
                     newItem.SubItems.Add(record.data.Length.ToString());
                     newItem.SubItems.Add(record.extraInfo);
+                    // This one requires special handling and cannot use function.
+                    newItem.SubItems.Add(record.opcodes[0].ToString("X").Substring(4, 4));
```
This throws an exception on header-only records and causes app to crash. Please add a check to make sure record.opcodes.Count > 0, else make this an empty string.

tfarley commented on aclogview/Program.cs:
```diff
@@ -5,6 +5,29 @@
 using System.Windows.Forms;
 
 namespace aclogview {
+    using Heading;   
```
What is "Heading"? Can this be deleted?

tfarley commented on aclogview/CM_Magic.cs:
```diff
@@ -111,7 +113,8 @@ public class CastUntargetedSpell : Message {
             TreeNode rootNode = new TreeNode(this.GetType().Name);
             rootNode.Expand();
-            rootNode.Nodes.Add("i_spell_id = " + i_spell_id);
+            rootNode.Expand();
```
Shouldn't expand twice, line can be deleted.

tfarley commented on aclogview/CM_Magic.cs:
```diff
@@ -127,8 +130,7 @@ public class RemoveSpell : Message {
             TreeNode rootNode = new TreeNode(this.GetType().Name);
-            rootNode.Expand();
-            rootNode.Nodes.Add("i_spell_id = " + i_spell_id);
+            rootNode.Nodes.Add("i_spell_id = " + Utility.FormatGuid((uint)this.i_spell_id));
```
You deleted the node expansion call above this. Please add it back - otherwise it's annoying to have to manually expand such simple nodes in the UI.

tfarley commented on aclogview/CM_Magic.cs:
```diff
@@ -93,8 +95,8 @@ public class CastTargetedSpell : Message {
             rootNode.Expand();
-            rootNode.Nodes.Add("i_target = " + i_target);
-            rootNode.Nodes.Add("i_spell_id = " + i_spell_id);
+            rootNode.Nodes.Add("i_target" + Utility.FormatGuid(this.i_target));
```
You killed the = here, please add it back to be consistent.

ogmage78 commented:
You have a great eye for catching bugs!   I never went back to test on the original method - I always use the new open as message.   I never saw that crash.   Thanks again and I think I have made all of the corrections you requested.

tfarley commented:
LGTM. Thanks!

ogmage78 commented:
Thanks!
